app/src/main/java/de/csicar/ning/Model.kt

```diff
@@ -39,20 +49,76 @@ data class DeviceWithName(
     val asDevice = Device(deviceId, networkId, ip, deviceName, hwAddress, isScanningDevice)
 
     val icon
-        get() = when (vendorName) {
-            "Espressif Inc." -> R.drawable.ic_memory_white_48dp
-            "ADMTEK INCORPORATED" -> R.drawable.ic_baseline_settings_ethernet_48
-            "LG Electronics (Mobile Communications)" -> R.drawable.ic_baseline_phone_android_48
-            "Nintendo Co.,Ltd" -> R.drawable.ic_baseline_videogame_asset_48
-            "AzureWave Technology Inc." -> R.drawable.ic_baseline_cast_48
-            "Compal Broadband Networks, Inc." -> R.drawable.ic_baseline_router_48
-            "Ubiquiti Networks Inc." -> R.drawable.ic_baseline_router_48
-            "Sonos, Inc." -> R.drawable.ic_baseline_speaker_48
-            "AVM Audiovisuelles Marketing und Computersysteme GmbH" -> R.drawable.ic_baseline_router_48
-            "TP-LINK TECHNOLOGIES CO.,LTD." -> R.drawable.ic_baseline_router_48
-            "HUAWEI TECHNOLOGIES CO.,LTD" -> R.drawable.ic_baseline_phone_android_48
-            "Xiaomi Communications Co Ltd" -> R.drawable.ic_baseline_phone_android_48
-            else -> R.drawable.ic_laptop_white_48dp
+        get() = when (deviceTypeEnum) {
```

Comment:
I think we could move `icon` and `deviceType` attribute to the `DeviceType` enum. (I did not think of that before hand, sorry)

Reply:
Besides that I think this is ready to merge :+1:

Reply:
Yes, good idea. I changed it.